node_modules/oae-principals/lib/api.user.js

```diff
@@ -774,3 +811,138 @@ var _setAdmin = function(ctx, adminType, isAdmin, principalId, callback) {
         return PrincipalsDAO.setAdmin(adminType, isAdmin, principalId, callback);
     });
 };
+
+/**
+ * Send an email token to a user that can be used to verify the user owns the email address
+ *
+ * @param  {Context}    ctx             Standard context object containing the current user and the current tenant
+ * @param  {User}       user            The user to send the email token to
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occurred, if any
+ * @api private
+ */
+var _sendEmailToken = function(ctx, user, callback) {
+    callback = callback || function(err) {
+        if (err) {
+            log().error({'err': err, 'userId': user.id}, 'Unable to send a user a verification email');
+        }
+    };
+
+    // Generate a token that will be sent to the user's email address
+    var token = util.format('email-%s', _generateEmailToken());
+
+    // Store the token
+    PrincipalsDAO.storeEmailToken(token, user.id, function(err) {
+        if (err) {
+            return callback(err);
+        }
+
+        // The EmailAPI expects a user to have a verified email address. As this is not the case
+        // when sending an email token, we send in a patched user object
+        var userToEmail = _.extend({}, user, {'email': user.emailChange});
+
+        // Send an email to the specified e-mail address
+        var data = {
+            'tenant': ctx.tenant(),
+            'user': userToEmail,
+            'baseUrl': TenantsUtil.getBaseUrl(ctx.tenant()),
+            'skinVariables': require('oae-ui').getTenantSkinVariables(ctx.tenant().alias),
+            'token': token
+        };
+        EmailAPI.sendEmail('oae-principals', 'verify', userToEmail, data, {'hash': token, 'ignoreVerification': true}, callback);
+    });
+};
+
+/**
+ * Resend an email token
+ *
+ * @param  {Context}    ctx             Standard context object containing the current user and the current tenant
```

Comment:
Just a general thought, but this now contains an authentication strategy, and a locale, and imposter, etc... Maybe rather than having the documentation denormalized, we should at some point switch it to "Standard request context". Which is shorter and more accurate.